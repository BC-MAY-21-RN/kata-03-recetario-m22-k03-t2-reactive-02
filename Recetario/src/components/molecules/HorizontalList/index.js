/* eslint-disable prettier/prettier */
import React from 'react';
import {ScrollView, StyleSheet, Text, View} from 'react-native';
import ListItem from '../../atoms/ListItem/listItem';

const HorizontalList = props => {
  const pantalla = props.pantalla;
  const productos = props.productos;
  return (
    <View style={styles.list}>
      <Text style = { styles.textPrincipal } >{ props.title }</Text>
        <ScrollView horizontal={true}>
        {productos.map((item, key) => (
            <ListItem key={key} title={item.title} id={item.id} url={item.img} pantalla={pantalla} />
        ))}
        </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  list: {
    marginTop: 20,
    marginBottom: 20,
  },
  textPrincipal: {
    textAlign: 'left',
    fontSize:25,
    margin: 3,
    color: '#ad1f61',
    paddingLeft: 5,
    fontWeight: 'bold',
  },
});

export default HorizontalList;
