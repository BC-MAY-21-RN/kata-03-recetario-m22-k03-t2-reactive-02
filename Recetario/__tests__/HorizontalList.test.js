import 'react-native';
import React from 'react';
import HorizontalList from '../src/components/molecules/HorizontalList';
import products from '../src/Productos/productos.json';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer
    .create(<HorizontalList productos={products} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
