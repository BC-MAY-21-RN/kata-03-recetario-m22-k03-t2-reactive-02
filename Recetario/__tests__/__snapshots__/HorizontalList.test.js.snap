// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "#ad1f61",
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingLeft": 10,
        "textAlign": "left",
      }
    }
  />
  <RCTScrollView
    horizontal={true}
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 200,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://t2.rg.ltmcdn.com/es/posts/2/7/7/lemon_pie_saludable_75772_600.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingTop": 5,
              "textAlign": "center",
            }
          }
        >
          Lemon Pie
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 200,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://t1.rg.ltmcdn.com/es/posts/1/3/8/budin_de_chocolate_con_aceite_75831_600.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingTop": 5,
              "textAlign": "center",
            }
          }
        >
          Chocolate Pudding
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 200,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://t2.rg.ltmcdn.com/es/posts/9/2/6/pastelitos_argentinos_75629_600.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingTop": 5,
              "textAlign": "center",
            }
          }
        >
          Argentinian Pastelitos
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 200,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://t2.rg.ltmcdn.com/es/posts/2/9/2/brownie_sin_harina_75292_600.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingTop": 5,
              "textAlign": "center",
            }
          }
        >
          Flourless Brownie 
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "height": 200,
            "margin": 5,
            "marginLeft": 10,
            "marginRight": 10,
            "marginTop": 15,
            "opacity": 1,
            "width": 130,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://t1.rg.ltmcdn.com/es/posts/0/4/5/pie_de_durazno_75540_600.jpg",
            }
          }
          style={
            Object {
              "borderRadius": 10,
              "height": 130,
              "width": 130,
            }
          }
        />
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "fontWeight": "bold",
              "paddingTop": 5,
              "textAlign": "center",
            }
          }
        >
          Peach Pie 
        </Text>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
